vars/transportRequestGetOrCreate.groovy

```diff
@@ -0,0 +1,78 @@
+import groovy.transform.Field
+
+import com.sap.piper.ConfigurationMerger
+import com.sap.piper.cm.ChangeManagement
+import com.sap.piper.cm.ChangeManagementException
+
+import hudson.AbortException
+
+
+@Field def STEP_NAME = 'transportRequestGetOrCreate'
+
+@Field Set parameterKeys = [
+    'changeDocumentId',
+    'developmentSystemId',
+    'credentialsId',
+    'endpoint'
+  ]
+
+@Field Set stepConfigurationKeys = [
+    'credentialsId',
+    'endpoint'
+  ]
+
+def call(parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        ChangeManagement cm = new ChangeManagement(script)
+
+        Map configuration = ConfigurationMerger.merge(parameters.script, STEP_NAME,
+                                                      parameters, parameterKeys,
+                                                      stepConfigurationKeys)
+
+        def changeDocumentId = configuration.changeDocumentId
+        if(!changeDocumentId) throw new AbortException('Change id not provided (parameter: \'changeDocumentId\').')
+
+        def developmentSystemId = configuration.developmentSystemId
+        if(!developmentSystemId) throw new AbortException('Development system id not provided (parameter: \'developmentSystemId\').')
+
+        def credentialsId = configuration.credentialsId
```

Comment:
The default for the credentialsId should be provided in the step specific default configuration shipped with the pipeline: ```resources/default_pipeline_environment.yml```. Alongside with the call to ```prepareDefaultValues``` it is ensured that this default (if not overwritten, of course) is always present.